devlib/trace/ftrace.py

```diff
@@ -103,6 +104,23 @@ def __init__(self, target,
                 raise TargetError('No trace-cmd found on device and no_install=True is specified.')
             self.target_binary = 'trace-cmd'
 
+        # Validate required events to be traced
+        available_events = self.target.execute(
+                'cat {}'.format(self.available_events_file)).splitlines()
+        selected_events = []
+        for event in self.events:
+            # Convert globs supported by FTrace into valid regexp globs
+            _event = event
+            if event[0] != '*':
+                _event = '*' + event
+            event_re = re.compile(_event.replace('*', '.*'))
+            # Select events matching the required ones
+            if len(filter(event_re.match, available_events)) == 0:
+                self.target.logger.warning('Event(s) [{}] not available for tracing'.format(event))
```

Comment:
This should be an exception rather than a warning -- if the client's request cannot be met, the client code should be infromed so it can handle it. (Perhaps add a `strict` flag to make the choice between ``exception``/``warning`` configurable)